lib/teselagen/api/test_client.py

```diff
@@ -340,12 +401,16 @@ def create_assay(
             "parserId": str(parser_id) if parser_id else None
         }
 
-        response: Dict[str, Any] = post(
-            url=self.create_assay_url.format(experiment_id),
-            headers=self.headers,
-            json=body,
-        )
-        # { id: "3" }
+        try:
+            response: Dict[str, Any] = post(
+                url=self.create_assay_url.format(experiment_id),
+                headers=self.headers,
+                json=body,
+            )
+        except Exception as e:
+            # TODO : Use a logger
+            raise Exception(e)
```

Comment:
I think a bare `raise` would propagate the exception, as [exemplified in the documentation](https://docs.python.org/3/tutorial/errors.html#raising-exceptions).

```suggestion
            raise
```